### PR TITLE
QoS-OLSR v2 Integration with MS v1.0

### PR DESCRIPTION
**Pull request by francislin12**
Qos_version_2

### CODE REVIEW

**Comment by mbaddeley** [comment]
Hi Francis, please confirm that all regression testing has been performed. I.e., with these changes does BATMAN work in the following configurations:

1. Laptop -> Eth -> Node -> BATMAN -> Node-> Eth -> Laptop
2. Laptop/Phone -> MCC AP -> Node -> BATMAN -> Node -> MCC AP -> Laptop/Phone
3. Laptop/Phone -> SCC AP -> Node -> BATMAN -> Node -> SCC AP -> Laptop/Phone

And then the same for QoS-OLSR:

4. Laptop -> Eth -> Node -> QoS-OLSR -> Node-> Eth -> Laptop
5. Laptop/Phone -> MCC AP -> Node -> QoS-OLSR -> Node -> MCC AP -> Laptop/Phone
6. Laptop/Phone -> SCC AP -> Node -> QoS-OLSR -> Node -> SCC AP -> Laptop/Phone

For the MCC/SCC modes it would be ok to use either Phone or Laptop.

**Review by mbaddeley — CHANGES_REQUESTED** [review]
Please convert all whitespace to the original format (x2 SPACES instead of x1 TAB) as this will make the merge a lot easier to read. Currently it is showing changes in almost every line because you have changed the whitespace format.

Here is how to change the vscode settings to show whitespace: https://stackoverflow.com/questions/30140595/show-whitespace-characters-in-visual-studio-code

Here is how to convert all the TABS back to spaces (make sure it converts to x2 spaces): https://stackoverflow.com/questions/33974681/how-can-i-convert-tabs-to-spaces-and-vice-versa-in-an-existing-file

**Review by joenpera — CHANGES_REQUESTED** [review]
Why platform detection was removed?  It is needed for correct ethernet port setup and it it different between hardwares.

File: modules/utils/docker/entrypoint.sh [hunk 1]
```diff
@@ -1,57 +1,157 @@
-#! /bin/bash
+#! /bin/bash -x
```
Comment (mbaddeley) [inline modules/utils/docker/entrypoint.sh:1]: remove -x

File: modules/utils/docker/entrypoint.sh [hunk 2]
```diff
@@ -1,57 +1,157 @@
...
-    mode=$CONCURRENCY
-    ch=$MCC_CHANNEL
-    ipaddr=$IP
-    nmask=$MASK
-    cc=$COUNTRY
-    psk=$KEY
-    txpwr=$TXPOWER
-    algo=$ROUTING
-    mesh_if=$MESH_VIF
+  source /opt/mesh.conf
```
Comment (mbaddeley) [inline modules/utils/docker/entrypoint.sh:125]: correct whitespace

File: modules/utils/docker/entrypoint.sh [hunk 3]
```diff
@@ -1,57 +1,157 @@
...
-   ifconfig br-lan down
-   brctl delbr br-lan
-   ifname_ap="$(ifconfig -a | grep wlan* | awk -F':' '{ print $1 }')"
-   ifname_mp="$(ifconfig -a | grep wlp1* | awk -F':' '{ print $1 }')"
-   ifconfig $ifname_ap down
-   ifconfig $ifname_mp down
-   ifconfig $ifname_ap up
-   ifconfig $ifname_mp up
-   exit 1
+  killall hostapd
```
Comment (mbaddeley) [inline modules/utils/docker/entrypoint.sh:140]: correct whitespace

File: modules/utils/docker/entrypoint.sh [hunk 4]
```diff
@@ -1,57 +1,157 @@
...
+  elif [ "$1" -ge 2412 ] && [ "$1" -le 2472 ]; then
+    retval_band="g"
+    retval_channel=$((("$1"-2407)/5))
+  else
+    echo "ERROR! frequency out of range!"
+    exit 1
+  fi
+}
+
+create_ap_config()
```
Comment (mbaddeley) [inline modules/utils/docker/entrypoint.sh:24]: Pretty sure this function is no longer used? @joenpera?

Comment (mbaddeley) [inline modules/utils/docker/entrypoint.sh:24]: Moved to a new place unecessarily. If you look at the original file it was originally down below.

File: modules/utils/docker/entrypoint.sh [hunk 5]
```diff
@@ -92,41 +182,54 @@ if [ "$mode" = "sta+mesh" ]; then
 elif [ "$mode" = "ap+mesh_mcc" ]; then
   # Create bridge br-lan
   brctl addbr br-lan
-  ifname_ap="$(ifconfig -a | grep wlan* | awk -F':' '{ print $1 }')"
+  ifname_ap="$(ifconfig -a | grep -E '^wlan' -m 1 | awk -F':' '{ print $1 }')"
```
Comment (mbaddeley) [inline modules/utils/docker/entrypoint.sh:185]: Unecessary change from develop

File: modules/utils/docker/entrypoint.sh [hunk 6]
```diff
@@ -92,41 +182,54 @@ if [ "$mode" = "sta+mesh" ]; then
...
+
   # Start AP
   /usr/sbin/hostapd -B /var/run/hostapd.conf -f /tmp/hostapd.log
   # Bridge AP and Mesh
   if [ "$algo" = "olsr" ]; then
-        brctl addif br-lan "$mesh_if" "$ifname_ap"
-        iptables -A FORWARD --in-interface $mesh_if -j ACCEPT
-        killall olsrd 2>/dev/null   
-        (olsrd -i br-lan -d 0)&
+    brctl addif br-lan "$mesh_if" "$ifname_ap"
```
Comment (mbaddeley) [inline modules/utils/docker/entrypoint.sh:225]: Fix whitespace

File: modules/utils/docker/entrypoint.sh [hunk 7]
```diff
@@ -181,52 +285,67 @@ elif [ "$mode" == "ap+mesh_scc" ]; then
...
+    wpa=2
+    wpa_key_mgmt=WPA-PSK
+    rsn_pairwise=CCMP
+    wpa_passphrase=$psk
+
+    wmm_enabled=1
+    beacon_int=1000
+
+    ### IEEE 802.11n
+    ieee80211n=0
```
Comment (joenpera) [inline modules/utils/docker/entrypoint.sh:312]: disable ieee80211n?

**Pull request closed without merging**